src/Dropdown.jsx

```diff
@@ -1,40 +1,48 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import * as Select from '../lib/cockpit-components-select.jsx';
 
-// TODO
-const handleClick = (event) => {
-    if (event.button !== 0)
-        return;
-    var action = this.props.actions[event.currentTarget.getAttribute('data-value')];
-    if (!action.disabled && action.onActivate)
-        action.onActivate();
-};
+class Dropdown extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            show: false,
+            title: this.props.actions[0].label
+        };
 
-const DropDown = (props) => {
-    return (
-        <div className="btn-group">
-            <button className="btn btn-default" type="button" data-value="0" onClick={handleClick}>
-                <span>{props.actions[0].label}</span>
-            </button>
-            <button className="btn btn-default dropdown-toggle" data-toggle="dropdown">
-                <div className="caret" />
-            </button>
-            <ul className="dropdown-menu dropdown-menu-right" role="menu">
-                {
-                    props.actions.map((action, index) => {
-                        return (
-                            <li key={index} className={ action.disabled ? 'disabled' : '' }>
-                                <a data-value={index} role="link" tabIndex="0" onClick={handleClick}>{action.label}</a>
-                            </li>
-                        );
-                    })
-                }
-            </ul>
-        </div>
-    );
-};
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    // TODO
+    handleClick (event) {
+
+    }
+
+    render() {
+        const menuItems = this.props.actions.map((action, index) => {
+            return (
+                <Select.SelectEntry
+                    data={action.label}
+                    key={index}
+                    disabled={action.disabled}
+                >
+                    {action.label}
+                </Select.SelectEntry>
+            );
+        });
+
+        return (
+            <div className="btn-group" >
+                <Select.Select initial={this.state.title} onChange={(event) => this.handleClick(event)}>
+                    {menuItems}
+                </Select.Select>
+            </div>
+        );
+    }
+}
```

Comment:
I really wonder if it's that beneficial to wrap an already existing `Select` React component into this new one -- there's really no extra logic here, just some glue.

Reply:
looks like ```div``` can keep the distance between this dropdown button and the commit button in its left. Delete this div now?